src/action_runner/action_runner.cpp

```diff
@@ -103,11 +103,26 @@ bool install_new_version_stage_1(const bool must_restart = false)
     }
 }
 
-bool install_new_version_stage_2(std::wstring_view installer_path, std::wstring_view install_path, const bool launch_powertoys)
+bool install_new_version_stage_2(std::wstring installer_path, std::wstring_view install_path, const bool launch_powertoys)
 {
-    if (MsiInstallProductW(installer_path.data(), nullptr) != ERROR_SUCCESS)
+    std::transform(begin(installer_path), end(installer_path), begin(installer_path), ::towlower);
+
+    if (installer_path.ends_with(L".msi"))
     {
-        return false;
+        if (MsiInstallProductW(installer_path.data(), nullptr) != ERROR_SUCCESS)
+        {
+            return false;
+        }
+    }
+    else
+    {
+        // If it's not .msi, then it's our .exe installer
+        SHELLEXECUTEINFOW sei{ sizeof(sei) };
+        sei.fMask = { SEE_MASK_FLAG_NO_UI | SEE_MASK_NOASYNC };
+        sei.lpFile = installer_path.c_str();
+        sei.nShow = SW_SHOWNORMAL;
+
+        return ShellExecuteExW(&sei) == TRUE;
     }
```

Comment:
Perhaps the code after this if-else block should be moved inside the if, it's a little bit confusing like this